### PR TITLE
[Console] Fix docblock Command->addOption

### PR DESCRIPTION
**Pull request by MarcHagen**
| Q             | A
| ------------- | ---
| Branch?       | all supported releases and master 
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | - 
| License       | MIT
| Doc PR        | -

PHPStorm is parsing docblocks for codehinting. Because `$default` is a mixed field, it parses the codeblock.
```
Expected parameter of type '\Symfony\Component\Console\Command\mixed|\Symfony\Component\Console\Command\The|void', 'string' provided
```
Becuase there are no type for each `@param` it assumes the `The` is a type/class that can be used. 
Hence the `\Symfony\Component\Console\Command\The` suggestion

I've seached in the git history why these are removed, and came out on symfony/symfony#41912

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Please report the issue to phpstorm, these annotations should be accepted.

File: src/Symfony/Component/Console/Command/Command.php [hunk 1]
```diff
@@ -429,9 +429,9 @@ public function addArgument(string $name, int $mode = null, string $description
     /**
      * Adds an option.
      *
-     * @param $shortcut The shortcuts, can be null, a string of shortcuts delimited by | or an array of shortcuts
-     * @param $mode     The option mode: One of the InputOption::VALUE_* constants
-     * @param $default  The default value (must be null for InputOption::VALUE_NONE)
+     * @param string|array|null         $shortcut The shortcuts, can be null, a string of shortcuts delimited by | or an array of shortcuts
```
Comment (stof) [inline src/Symfony/Component/Console/Command/Command.php:432]: should be `string|string[]|null` in phpdoc, to also document the type of array items

**Comment by stof** [comment]
@nicolas-grekas phpstan does not accept them either

File: src/Symfony/Component/Console/Command/Command.php [hunk 2]
```diff
@@ -429,9 +429,9 @@ public function addArgument(string $name, int $mode = null, string $description
...
+     * @param int|null                  $mode     The option mode: One of the InputOption::VALUE_* constants
```
Comment (stof) [inline src/Symfony/Component/Console/Command/Command.php:433]: our description is about wrong here. It is a bitmask of those constants.

**Comment by nicolas-grekas** [comment]
That's a bug in phpstan then, the spec makes the type optional:
https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/param.html

**Comment by MarcHagen** [comment]
Well... changing the base of the PR was not that smart... sorry about that.

File: src/Symfony/Component/Console/Command/Command.php (hunk 2 above)
Comment (MarcHagen) [reply src/Symfony/Component/Console/Command/Command.php:433]: Well the `InputOption->__construct` docblock is the same / more different that is listed here.
https://github.com/symfony/symfony/blob/137aa3ba3c73e823e8259352beaae66d25f13def/src/Symfony/Component/Console/Input/InputOption.php#L55-L62

**Comment by MarcHagen** [comment]
> Please report the issue to phpstorm, these annotations should be accepted.

Yes quick search revealed that the issue from 2018, which is not all that good either because it assumes the type is mandatory
https://youtrack.jetbrains.com/issue/WI-16082

**Pull request closed without merging**